core/src/main/java/discord4j/core/object/entity/User.java

```diff
@@ -97,19 +108,37 @@ public final String getUsername() {
      * Gets the user's 4-digit discriminator
      * The discriminator is unique number to distinct one among all users with the same username.
      * The discriminator is randomly generated, but can be changed if the user has a nitro subscription.
+     * Migrated users from the old system will have a discriminator of "0".
+     * May become null after the system change is complete.
      *
-     * @return The user's 4-digit discriminator.
+     * @return The user's 4-digit discriminator, or "0" if the user is migrated to the new system.
+     * @deprecated This method will be removed once the system change is complete.
      */
+    @Deprecated
+    @Nullable
     public final String getDiscriminator() {
         return data.discriminator();
     }
 
     /**
-     * Gets the user's username and discriminator separated by a #
-     * This is unique across the discord platform, but may change.
-     * @return {@link User#getUsername()}#{@link User#getDiscriminator()}
+     * Returns whether the user is migrated to the new system or not.
+     *
+     * @return true if the user is migrated to the new system
+     */
+    private boolean isMigrated() {
+        return getDiscriminator() == null || getDiscriminator().equals("0");
+    }
+
+    /**
+     * Gets the user's username and discriminator separated by a # or its username if the user is using the new system
+     *
+     * @return {@link User#getUsername()}#{@link User#getDiscriminator()} if the user is not migrated, {@link User#getUsername()} otherwise.
      */
     public final String getTag() {
+        if (isMigrated()) {
+            return getUsername();
```

Comment:
I think this should be `return getGlobalName();`

Reply:
For client-side discord allow use the two forms for tag but if you use \ for get the "raw" show the username and not the global_name

Reply:
Ah Thanks for clarifying it, I misread and was confusing global with unique